### ISSUE TITLE
Jvm test suite type should not be unique

### ISSUE THREAD

**Issue by xenoterracide** [gradle/gradle#30446]
### Expected Behavior

should be valid so that plugins like the jacoco aggregate report can aggreate both.

```kts
   val test by getting(JvmTestSuite::class) {
      testType = TestSuiteType.UNIT_TEST
      dependencies {
        implementation(libs.spring.orm)
        compileOnly(libs.jspecify)
      }
    }
    val whitebox by registering(JvmTestSuite::class) {
      testType = TestSuiteType.UNIT_TEST
```

### Current Behavior (optional)

```
* What went wrong:
Could not determine the dependencies of task ':controller-registration:compileJava'.
> Could not resolve all dependencies for configuration ':controller-registration:compileClasspath'.
   > Could not resolve project :jpa.
     Required by:
         project :controller-registration > project :model-security
      > Could not configure suite: 'whitebox'. Another test suite: 'test' uses the type: 'unit-test' and has already been configured in project: 'jpa'.
```

### Context

Because I use JPMS and follow the [test trophy](https://martinfowler.com/articles/2021-test-shapes.html) model, I sometimes need to create two test suites with similar profiles. While unit testing is my example, I primarily write integration tests. The challenge is that my "blackbox by default" test suites don't provide access to package-private methods, forcing me to switch to classpath testing.

I can also see this approach being useful outside of JPMS. For instance, when integrating a library with different implementations, such as testing Spring Data JPA with both EclipseLink and Hibernate. There are likely other scenarios where this method is needed.

I could use custom strings for the suite names, but what happens if I have both whiteboxUnit and whiteboxIntegration? This approach can quickly become excessive, especially with examples like JPA. It still wouldn't solve the issue with the JaCoCo aggregation plugin, which only aggregates a single test type. Additional undocumented steps would be necessary to make it work.

Moreover, this method could be confusing for someone new to the project. Renaming the suite doesn't truly create a new type; it just adds another label. At that point, you might as well stick with the suite name. Another option would be to use a different, albeit inaccurate, test type—but again, the aggregation plugin doesn’t handle this out of the box.

```
------------------------------------------------------------
Gradle 8.10
------------------------------------------------------------

Build time:    2024-08-14 11:07:45 UTC
Revision:      fef2edbed8af1022cefaf44d4c0514c5f89d7b78

Kotlin:        1.9.24
Groovy:        3.0.22
Ant:           Apache Ant(TM) version 1.10.14 compiled on August 16 2023
Launcher JVM:  21.0.4 (Eclipse Adoptium 21.0.4+7-LTS)
Daemon JVM:    /home/xeno/.asdf/installs/java/temurin-21.0.4+7.0.LTS (no JDK specified, using current Java home)
OS:            Linux 6.6.44-1-MANJARO amd64
```

**Comment by xenoterracide** [comment]
loosely related #30445

**Comment by xenoterracide** [comment]
> However, we believe that the original problem @xenoterracide is experiencing is that Gradle can't aggregate multiple reports of the same type from a single project. @xenoterracide can you confirm?

Yes, that's the only *functional* issue I've encountered so far, though it's not necessarily the only one. Another point to consider is that these settings could serve as "code as documentation." The test type should imply the type of the test suite, not just its name. I believe the domain model—specifically for test suites—is incorrectly represented, which might be the root of the issue.

As it stands, I think the current `testType` should be renamed to `suiteName` or simply `name`, while `testType` or `suiteType` should reflect what `TestSuiteType` holds now. This would still allow for arbitrary strings while making the distinction clearer. That way, the JaCoCo report aggregation would still rely on the type, but the suite name would be more accurately labeled.

You could even make this change without altering the current ABI, so it wouldn’t break any existing code. Instead, you could simply announce the semantic change.

**Comment by tresat** [comment]
We'll be thinking more about testing and test suites soon as part of Declarative Gradle modeling efforts, and will be sure to consider these points.  Thanks for filing this!

**Comment by jvandort** [comment]
Closing since test suite type no longer exists after https://github.com/gradle/gradle/pull/31706
